lib.php

```diff
@@ -96,7 +96,7 @@ function theme_boost_union_get_main_scss_content($theme) {
     global $CFG;
 
     $scss = '';
-    $filename = !empty($theme->settings->preset) ? $theme->settings->preset : null;
+    $filename = get_config('theme_boost_union', 'preset');
```

Comment:
The goal of this PR is to replace `$theme->settings` with `get_config('theme_boost_union', ...)`.
This is fine. However, as this part of the function is duplicated from upstream Boost Core and to avoid any hickups with edge cases, I would like to ask you to keep the rest of the logic in this line unchanged unless you prove me that it must be changed.

Reply:
get_config($plugin,$value) returns either false, true or a string (which may be empty).

I removed !empty and the rest mainly because it is not necessary to set $filename explicitly to null, because `if ($filename)` returns false when $filename is null, false or an empty string, which is the desired behaviour.

But you're right, it would be easier when updating the upstream code to only replace that part.

By the way, looking at the function again, couldn't you, too, just reuse Boost's settings by doing this:

```
function theme_boost_union_get_main_scss_content($theme) {
    global $CFG;

    $scss = file_get_contents($CFG->dirroot . '/theme/boost_union/scss/boost_union/pre.scss');
    $scss .= theme_boost_get_main_scss_content(theme_config::load('theme_boost'));
    $scss .= file_get_contents($CFG->dirroot . '/theme/boost_union/scss/boost_union/post.scss');
    return $scss;
}
```

tl;dr
I'll change the code to leave the stuff in.
But there could be another way...?